From 388825c73ec576327ac3d5b00f0d2765bc54f91f Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 03 Mar 2022 16:02:01 +0200
Subject: [PATCH] [SW-66329] gaudi3: device: pci: add die1 regs

Added pci registers addresses init for die1 of gaudi3, in order
to use them in the pif arc binary which is intended to run on
both dies.

Change-Id: Ibefdd28706234d88114b87db05c531480db77ce9
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 534b076..1098a55 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -18,6 +18,13 @@
 #include <d0_parc_arc0_cfg_regs.h>
 #include <d0_psoc_i2c_s_regs.h>
 #include <d0_parc_int_aggr_i2c_slv_comb_regs.h>
+#include <d0_parc_local_cfg_regs.h>
+
+#include <d1_pcie_aux_regs.h>
+#include <d1_pcie_core_regs.h>
+#include <d1_pcie_dbi_sig_regs.h>
+
+
 #include <gaudi3/gaudi3_boot_info.h>
 
 #include <bitops.h>
@@ -205,6 +212,15 @@
 	image_entry();
 }
 
+static u32 hl_get_die_id(void)
+{
+	/*
+	 * same address space read by both dies will return a different value
+	 * depending on the die location
+	 */
+	return regmap_readl(mmD0_PARC_LOCAL_CFG_DIE_ID);
+}
+
 #define GAUDI3_EFUSE_READ_RANGE_ALL	0xff00
 
 void device_init_efuse(void)
@@ -226,21 +242,39 @@
 
 void device_init_pci(void)
 {
-	pci.dbi_bar0_reg = mmD0_PCIE_DBI_SIG_BAR0_REG;
-	pci.dbi_bar2_reg = mmD0_PCIE_DBI_SIG_BAR2_REG;
-	pci.dbi_bar4_reg = mmD0_PCIE_DBI_SIG_BAR4_REG;
-	pci.dbi_status_command_reg = mmD0_PCIE_DBI_SIG_STATUS_COMMAND_REG;
-	pci.dbi_uncorr_aer_mask = mmD0_PCIE_DBI_SIG_UNCORR_ERR_MASK_OFF;
-	pci.aux_dbi_reg = mmD0_PCIE_AUX_DBI;
-	pci.aux_mem_space_en_reg = mmD0_PCIE_AUX_MEM_SPACE_EN;
-	pci.aux_flr_ctrl_reg = mmD0_PCIE_AUX_FLR_CTRL;
-	pci.aux_ltssm_en_reg = mmD0_PCIE_AUX_LTSSM_EN;
-	pci.aux_pf_flr_active_reg = mmD0_PCIE_AUX_PF_FLR_ACTIVE;
-	pci.aux_flr_int_reg = mmD0_PCIE_AUX_FLR_INT;
-	pci.aux_pf_flr_done_reg = mmD0_PCIE_AUX_PF_FLR_DONE;
-	pci.aux_rdlh_link_up_reg = mmD0_PCIE_AUX_RDLH_LINK_UP;
-	pci.aux_smlh_link_up_reg = mmD0_PCIE_AUX_SMLH_LINK_UP;
-	pci.core_hot_rst_reg = mmD0_PCIE_CORE_HOT_RST;
+	if (!hl_get_die_id()) {	/* DIE0 */
+		pci.dbi_bar0_reg = mmD0_PCIE_DBI_SIG_BAR0_REG;
+		pci.dbi_bar2_reg = mmD0_PCIE_DBI_SIG_BAR2_REG;
+		pci.dbi_bar4_reg = mmD0_PCIE_DBI_SIG_BAR4_REG;
+		pci.dbi_status_command_reg = mmD0_PCIE_DBI_SIG_STATUS_COMMAND_REG;
+		pci.dbi_uncorr_aer_mask = mmD0_PCIE_DBI_SIG_UNCORR_ERR_MASK_OFF;
+		pci.aux_dbi_reg = mmD0_PCIE_AUX_DBI;
+		pci.aux_mem_space_en_reg = mmD0_PCIE_AUX_MEM_SPACE_EN;
+		pci.aux_flr_ctrl_reg = mmD0_PCIE_AUX_FLR_CTRL;
+		pci.aux_ltssm_en_reg = mmD0_PCIE_AUX_LTSSM_EN;
+		pci.aux_pf_flr_active_reg = mmD0_PCIE_AUX_PF_FLR_ACTIVE;
+		pci.aux_flr_int_reg = mmD0_PCIE_AUX_FLR_INT;
+		pci.aux_pf_flr_done_reg = mmD0_PCIE_AUX_PF_FLR_DONE;
+		pci.aux_rdlh_link_up_reg = mmD0_PCIE_AUX_RDLH_LINK_UP;
+		pci.aux_smlh_link_up_reg = mmD0_PCIE_AUX_SMLH_LINK_UP;
+		pci.core_hot_rst_reg = mmD0_PCIE_CORE_HOT_RST;
+	} else {	/* DIE1 */
+		pci.dbi_bar0_reg = mmD1_PCIE_DBI_SIG_BAR0_REG;
+		pci.dbi_bar2_reg = mmD1_PCIE_DBI_SIG_BAR2_REG;
+		pci.dbi_bar4_reg = mmD1_PCIE_DBI_SIG_BAR4_REG;
+		pci.dbi_status_command_reg = mmD1_PCIE_DBI_SIG_STATUS_COMMAND_REG;
+		pci.dbi_uncorr_aer_mask = mmD1_PCIE_DBI_SIG_UNCORR_ERR_MASK_OFF;
+		pci.aux_dbi_reg = mmD1_PCIE_AUX_DBI;
+		pci.aux_mem_space_en_reg = mmD1_PCIE_AUX_MEM_SPACE_EN;
+		pci.aux_flr_ctrl_reg = mmD1_PCIE_AUX_FLR_CTRL;
+		pci.aux_ltssm_en_reg = mmD1_PCIE_AUX_LTSSM_EN;
+		pci.aux_pf_flr_active_reg = mmD1_PCIE_AUX_PF_FLR_ACTIVE;
+		pci.aux_flr_int_reg = mmD1_PCIE_AUX_FLR_INT;
+		pci.aux_pf_flr_done_reg = mmD1_PCIE_AUX_PF_FLR_DONE;
+		pci.aux_rdlh_link_up_reg = mmD1_PCIE_AUX_RDLH_LINK_UP;
+		pci.aux_smlh_link_up_reg = mmD1_PCIE_AUX_SMLH_LINK_UP;
+		pci.core_hot_rst_reg = mmD1_PCIE_CORE_HOT_RST;
+	}
 }
 
 void device_init_i2c_s(void)
